File: src/main/java/red/tetracube/smarthub/iot/payloads/DeviceProvisioning.java
package red.tetracube.smarthub.iot.payloads;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;
import java.util.UUID;

public record DeviceProvisioning(
        @JsonProperty("id") UUID id,
        @JsonProperty("name") String name,
        @JsonProperty("feedback_topic") String feedbackTopic,
        @JsonProperty("features") List<FeatureProvisioning> features
) {
}

File: src/main/kotlin/iot/tetracubered/smarthub/messaging/payloads/DeviceProvisioning.kt
package iot.tetracubered.smarthub.messaging.payloads

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class DeviceProvisioning @JsonCreator constructor(
    @JsonProperty("id") @field:NotNull val circuitId: UUID,
    @JsonProperty("default_name") @field:NotNull val name: String,
    @JsonProperty("feedback_topic") @field:NotNull val feedbackTopic: String,
    @JsonProperty("actions") @field:NotNull @field:NotEmpty val actions: List<DeviceActionProvisioning>
)